src/tests/ftest/util/command_utils_base.py

```diff
@@ -524,6 +524,10 @@ def _get_new(self):
         """
         return CommandWithParameters(self.namespace, self._command, self._path)
 
+    def command_prepend_il(self, libpath):
+        """Add LD_PRELOAD=libpath into command to enable interception library"""
+        self._command = f'LD_PRELOAD={libpath} {self._command}'
```

Comment:
This works for this specific test/command, but not generally. The main problem being that this doesn't account for running as `sudo`, `mpirun`, etc. And since this code is shared for many commands, it could cause confusion elsewhere if someone tries to use it.
For example, these are some different ways an env is exported for other commands:
```
export foo=bar; my_command
clush 'export foo=bar; my_command'
sudo foo=bar my_command
```
Because of this, we have the command and its environment separated so different ways of executing can handle this as needed.

Furthermore, we can't hardcode `/usr/lib64` because that isn't correct for source builds. We have to use `os.path.join(self.prefix, 'lib64', 'libpil4dfs.so')`
In your test, you should be able to just replace this:
```
self.fio_cmd.command_prepend_il('/usr/lib64/libpil4dfs.so')
```
with this:
```
self.fio_cmd.env['LD_PRELOAD'] = os.path.join(self.prefix, 'lib64', 'libpil4dfs.so')
```

And if that doesn't work, please let me know.

Reply:
I did try self.fio_cmd.env['LD_PRELOAD'] = os.path.join(self.prefix, 'lib64', 'libpil4dfs.so'). It seems libpil4dfs.so is not loaded.

Reply:
Do you have an example log for that? If that's not working, there's probably a bug somewhere in the core code

Reply:
Forgot to mention that the file is on boro.

Reply:
Thanks. I see the problem in the core code. Please just continue with your solution and I will create a ticket to fix the problem, and I will update your solution appropriately.

Reply:
Just FYI - https://daosio.atlassian.net/browse/DAOS-13143
Thanks for confirming the problem :) These things are easy to miss...